tests/unittests/benchmark/test_benchmark_client.py

```diff
@@ -0,0 +1,142 @@
+                        ],
+                        "task": [
+                            {
+                                "orion-benchmark-task-rosenbrock-RosenBrock": {
+                                    "dim": 3,
+                                    "max_trials": 25,
+                                }
+                            },
+                            {
+                                "orion-benchmark-task-carromtable-CarromTable": {
+                                    "max_trials": 20
+                                }
+                            },
+                        ],
+                    }
+                ],
+            }
+            assert bm1.configuration == cfg
+
+            assert bm1.configuration == bm2.configuration
+
+    def test_create_with_only_name(self):
+        """Test creation with a not existed benchmark name"""
+        with OrionState():
+            name = "bm00001"
+            with pytest.raises(NoConfigurationError) as exc:
+                get_or_create_benchmark(name)
+
+            assert "Benchmark {} does not exist in DB".format(name) in str(exc.value)
+
+    def test_create_with_invalid_algorithms(self):
+        """Test creation with a not existed algorithm"""
+        with OrionState():
+            name = "bm00001"
+            with pytest.raises(NotImplementedError) as exc:
+                get_or_create_benchmark(
+                    name,
+                    algorithms=[{"fake_algorithm": {"seed": 1}}],
+                    targets=[
+                        {
+                            "assess": [AverageResult(2), AverageRank(2)],
+                            "task": [RosenBrock(25, dim=3), CarromTable(20)],
+                        }
+                    ],
+                )
+            assert "Could not find implementation of BaseAlgorithm" in str(exc.value)
+
+    def test_create_with_invalid_targets(self, algorithms):
+        """Test creation with invalid Task and Assessment"""
+        with OrionState():
+            name = "bm00001"
+            with pytest.raises(AttributeError) as exc:
+                get_or_create_benchmark(
+                    name,
+                    algorithms=algorithms,
+                    targets=[{"assess": [AverageResult(2)], "task": [DummyTask]}],
+                )
+            print(str(exc.value))
+
+            assert "type object '{}' has no attribute ".format("DummyTask") in str(
+                exc.value
+            )
+
+            with pytest.raises(AttributeError) as exc:
+                get_or_create_benchmark(
+                    name,
+                    algorithms=algorithms,
+                    targets=[
+                        {"assess": [DummyAssess], "task": [RosenBrock(25, dim=3)]}
+                    ],
+                )
+            print(str(exc.value))
+
+            assert "type object '{}' has no attribute ".format("DummyAssess") in str(
+                exc.value
+            )
```

Comment:
The function `_resolve_db_config` looks tricky. I would test it carefully.

Reply:
by tricky,  do you mean the import module based on the string configure from db part?

Reply:
Sorry the comment wasn't clear at all. I mean that the comprehension of the function requires lots of efforts and therefore bugs could easily be introduced without us noticing it if the function is only tested superficially.